version: '3.8'

services:
  vh_app:
    image: python:3.10
    volumes:
      - ./voice_helper_app/src:/app
    env_file:
      - ./.env
    environment:
      PYTHONUNBUFFERED: 1
      PYTHONDONTWRITEBYTECODE: 1
    restart: unless-stopped
    networks:
      - movies_network
    command: >
      bash -c "pip install --upgrade pip
      && pip install --no-cache-dir -r /app/requirements.txt
      && python /app/main.py"
    ports:
      - "8001:8001"

  vh_mongo_db:
    image: mongo:6.0.5-jammy
    restart: always
    expose:
      - "27017"
    env_file:
      - ./.env
    volumes:
      - ./db/mongo:/docker-entrypoint-initdb.d
    healthcheck:
      test: [ "CMD", "mongo", "--eval", "db.runCommand('ping').ok" ]
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - "27017:27017"
    networks:
      - movies_network

  vh_mongo_express:
    image: mongo-express:1.0.0-alpha
    restart: always
    expose:
      - "8081"
    environment:
      ME_CONFIG_MONGODB_SERVER: vh_mongo_db
      ME_CONFIG_MONGODB_PORT: 27017
    ports:
      - "8081:8081"
    networks:
      - movies_network

networks:
  movies_network:
    external: true